### ISSUE TITLE
Incompatibility with ember-element-helper v0.6.1 and Embroider optimized

### ISSUE THREAD

**Issue by jelhan** [ember-animation/ember-animated#594]
**Describe the bug**

Ember Animated breaks the build if used with Embroider optimized configuration and `ember-element-helper@0.6.1`:

```
Missing helper: -element in node_modules/ember-animated/dist/components/animated-container.js
```

The issue is caused by trying to resolve a `-element` helper, which does not exist if using `ember-element-helper@0.6.1`.

The issue only exists if using Embroider optimized. Ember Animated already detects at run-time if `element` or `-element` helper is available. https://github.com/ember-animation/ember-animated/blob/8ce1a4dfea6ef69f3f608f2a2cfaa0043a5b61e4/addon/src/components/animated-container.hbs#L1-L19

Based on build-time configuration depending on used `ember-element-helper` version. https://github.com/ember-animation/ember-animated/blob/8ce1a4dfea6ef69f3f608f2a2cfaa0043a5b61e4/addon/src/components/animated-container.ts#L204-L206

Bit if using Embroider optimized, the correct helper must be chosen on build-time already. A run-time switch is too late as Embroider / Webpack tries to resolve the helper already at build-time if using `staticHelpers` configuration.

Downgrading to `ember-element-helper@0.6.0` solves the issue.

**To Reproduce**

I created a reproduction here: https://github.com/jelhan/test-ember-animated-with-embroider-optimized

The error described above is shown in CI pipeline of `main` branch.

The following PR shows that it could be solved by downgrading `ember-element-helper` to `0.6.0`.

**Additional context**

It seems that @SergeAstapov started working on a big fix in #455 and #429. But it went stale.

**Comment by cah-brian-gantzler** [comment]
Just ran in a very similar issue. This time it was ember-element-helper coming from https://github.com/cibernox/ember-power-select/ember-basic-downdown which brought in 0.6.1. When resolving to 0.6.0 code again works under embroider full compat mode.

**Comment by rogeraraujo90** [comment]
I'm seeing this issue in my project too. I'm updating from Ember 3.28 to Ember 4.4. Adding resolution to `ember-element-helper` is an effective workaround. Thanks, @jelhan

**Comment by NullVoxPopuli** [comment]
anyone want to give this a go? https://github.com/ember-animation/ember-animated/pull/633

**Comment by jelhan** [comment]
Isn't this fixed already? https://github.com/ember-animation/ember-animated/blob/6e6175c368a0ee0c386ff9d67e5ffe6bbfa99264/addon/package.json#L67

**Comment by NullVoxPopuli** [comment]
Nay, the issue is that -element doesn't exist as a helper and breaks statically analyzed builds.

**Comment by jelhan** [comment]
> Nay, the issue is that -element doesn't exist as a helper and breaks statically analyzed builds.

Should have read my own issue description more carefully. Thanks for clarifying.

It seems @SergeAstapov picked up  #455 again. That seems to be the correct path forward on case we want to support `ember-element-helper <= 6.0.0` and `ember-element-helper >= 6.1.0` in parallel. This would for sure ease upgrades to consumers. But if we run into issues we should just drop support for `ember-element-helper < 6.1.0` and move forward.

**Comment by SergeAstapov** [comment]
this should be fixed in [v2.0.0](https://github.com/ember-animation/ember-animated/releases/tag/v2.0.0)
